Graph-Convolutional-Networks-Model-Relations-In-Data/graph_convolutional_networks_model_relations_in_data.ipynb

```diff
@@ -0,0 +1,692 @@
+    "\n",
+    "with open(\"glove.6B.300d.txt\", 'r', encoding=\"utf-8\") as f:\n",
+    "    for line in f:\n",
+    "        values = line.split()\n",
+    "        word = values[0]\n",
+    "        vector = np.asarray(values[1:], \"float32\")\n",
+    "        embeddings_dict[word] = vector"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Calculate GloVe embeddings for each label in our target label subset.\n",
+    "small_labels = ['house', 'birds', 'sun', 'valley',\n",
+    "               'nighttime', 'boats', 'mountain', 'tree', 'snow', 'beach', 'vehicle', 'rocks',\n",
+    "               'reflection', 'sunset', 'road', 'flowers', 'ocean', 'lake', 'window', 'plants',\n",
+    "               'buildings', 'grass', 'water', 'animal', 'person', 'clouds', 'sky']\n",
+    "vectorised_labels = [embeddings_dict[label].tolist() for label in small_labels]\n",
+    "\n",
+    "# Save them for further use.\n",
+    "word_2_vec_path = 'word_2_vec_glow_classes.json'\n",
+    "with open(word_2_vec_path, 'w') as fp:\n",
+    "    json.dump({\n",
+    "        'vect_labels': vectorised_labels,\n",
+    "    }, fp, indent=3)\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Let's check how well GloVe represents label names from our dataset.\n",
+    "# It would be hard to visualize vectors with 300 values, but luckly we have t-SNE for that.\n",
+    "# This function builds a t-SNE model(https://www.learnopencv.com/t-sne-for-feature-visualization/) \n",
+    "# for label embeddings and visualizes them.\n",
+    "def tsne_plot(model):\n",
+    "    labels = []\n",
+    "    tokens = []\n",
+    "\n",
+    "    for word in model:\n",
+    "        tokens.append(model[word])\n",
+    "        labels.append(word)\n",
+    "    \n",
+    "    tsne_model = TSNE(perplexity=2, n_components=2, init='pca', n_iter=25000, random_state=2020, n_jobs=4)\n",
+    "    new_values = tsne_model.fit_transform(tokens)\n",
+    "\n",
+    "    x = []\n",
+    "    y = []\n",
+    "    for value in new_values:\n",
+    "        x.append(value[0])\n",
+    "        y.append(value[1])\n",
+    "        \n",
+    "    plt.figure(figsize=(13, 13)) \n",
+    "    for i in range(len(x)):\n",
+    "        plt.scatter(x[i],y[i])\n",
+    "        plt.annotate(labels[i],\n",
+    "                     xy=(x[i], y[i]),\n",
+    "                     xytext=(5, 2),\n",
+    "                     size=15,\n",
+    "                     textcoords='offset points',\n",
+    "                     ha='right',\n",
+    "                     va='bottom')\n",
+    "    plt.show()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Now we can draw t-SNE visualisation.\n",
+    "tsne_plot(dict(zip(small_labels, vectorised_labels)))"
```

Comment:
You form a dictionary here but then in the function you return back to two lists. Remove the unnecessary conversion.